tool_collections/bamtools/bamtools_split/bamtools-split.xml

```diff
@@ -1,108 +1,146 @@
-                                    option from the menu will allow you to enter the tag name. As was the
-                                    case with the reference splitting above, this option can produce very
-                                    large number of outputs if a tag has a large number of unique values.
-
------
-
-.. class:: infomark
-
-**More information**
-
-Additional information about BAMtools can be found at https://github.com/pezmaster31/bamtools/wiki
-
-    </help>
-    <citations>
-        <citation type="doi">10.1093/bioinformatics/btr174</citation>
-    </citations>
-</tool>
+<tool id="bamSplit" name="Split" version="2.4.1">
+    <description>BAM datasets on variety of attributes</description>
+    <macros>
+        <xml name="macro_output_type">
+            <param name="output_type" type="select" label="output as">
+                <option value="history_items">history items</option>
+                <option value="dataset_collection">dataset_collection</option>
+            </param>
+        </xml>
+    </macros>
+    <requirements>
+        <requirement type="package" version="2.4.0">bamtools</requirement>
+    </requirements>
+    <command>
+        <![CDATA[
+            echo "BAM" > $report &&
+            #for $bam_count, $input_bam in enumerate( $input_bams ):
+                ln -s "${input_bam}" "localbam_${bam_count}.bam" &&
+                ln -s "${input_bam.metadata.bam_index}" "localbam_${bam_count}.bam.bai" &&
+            #end for
+            bamtools
+            split
+            #if str ( $analysis_type.analysis_type_selector ) == "-tag" :
+                ${analysis_type.analysis_type_selector} "${analysis_type.tag_name}"
+            #else
+                ${analysis_type.analysis_type_selector}
+            #end if
+            -stub split_bam
+            #for $bam_count, $input_bam in enumerate( $input_bams ):
+                -in "localbam_${bam_count}.bam"
+            #end for
+            #if str ( $analysis_type.analysis_type_selector ) == "-reference"  and $analysis_type.output_type == "dataset_collection": 
+                #import re
+                #set $name = $re.sub('\W','_',$re.sub('\.bam$','',$input_bams[0].name))
+                #set $ref_list = ' '.join([$re.sub('^.*__sq__(.+)__sq__.*$','\\1',n) if n.find('__sq__') >= 0 else n for n in str($input_bam.metadata.reference_names).split(',')])
+                    && (export I=0; 
+                      for i in $ref_list; 
+                        do I=\$((++I)); SN=`printf "split_bam.REF_%s.bam" "\$i"`; 
+                           if [ -e \$SN ]; 
+                             then FN=`printf "split_bam%05d%s.%s.bam" \$((I)) "$name" "\$i"`; 
+                             mv \$SN \$FN;
+                           fi;
+                        done)
+            #end if
+        ]]>
+    </command>
+    <inputs>
+        <param name="input_bams" type="data" format="bam" label="BAM dataset(s) to filter" min="1" multiple="True"/>
```

Comment:
I'd consider making this just a single dataset input - this way if the users come with a dataset list the tool will produce a nested list naturally. I assume that would be the most common use case - ping @nekrut. If you still want to filter multiple individual bams into a flat list - I'd guess I'd make this a conditional but then you'd have to explain that operation and that wouldn't be easy.